--- ./src/libnet_init.c	2002-08-06 00:26:03.000000000 +0200
+++ ./src/libnet_init.c.oden	2008-12-16 16:52:44.000000000 +0100
@@ -84,7 +84,7 @@ libnet_init(int injection_type, char *de
             }
             if (libnet_open_link(l) == -1)
             {
-                snprintf(err_buf, LIBNET_ERRBUF_SIZE, l->err_buf);
+                snprintf(err_buf, LIBNET_ERRBUF_SIZE, "%s", l->err_buf);
                 goto bad;
             }
             break;
@@ -100,7 +100,7 @@ libnet_init(int injection_type, char *de
             }
             if (libnet_open_raw4(l) == -1)
             {
-                snprintf(err_buf, LIBNET_ERRBUF_SIZE, l->err_buf);
+                snprintf(err_buf, LIBNET_ERRBUF_SIZE, "%s", l->err_buf);
                 goto bad;
             }
             break;
@@ -116,7 +116,7 @@ libnet_init(int injection_type, char *de
             }
             if (libnet_open_raw6(l) == -1)
             {
-                snprintf(err_buf, LIBNET_ERRBUF_SIZE, l->err_buf);
+                snprintf(err_buf, LIBNET_ERRBUF_SIZE, "%s", l->err_buf);
                 goto bad;
             }
             break;
